blocks/editable/index.js

```diff
@@ -131,10 +131,18 @@ export default class Editable extends wp.element.Component {
 		}
 
 		const content = this.getContent();
+		const collapsed = this.editor.selection.isCollapsed();
 
 		this.setState( {
 			empty: ! content || ! content.length,
 		} );
+
+		if ( this.props.focus.collapsed !== collapsed ) {
```

Comment:
It think this is possible when the editor become active (before the rerender), it's worth checking IMO. good catch @aduth